Add local binding command and examples for kv and DO (#3142)
cloudflare/cloudflare-docs: products/pages/src/content/platform/functions.md

## Adding bindings

> While bringing your Workers to Pages, bindings are a big part of what makes your application truly full-stack. You can add KV, Durable Object, and plain-text bindings to your project. You can also use these bindings in development with [wrangler](https://developers.cloudflare.com/pages/platform/functions#develop-and-preview-locally). 

Reviewer: ```suggestion
While bringing your Workers to Pages, bindings are a big part of what makes your application truly full-stack. You can add KV, Durable Object, and plain-text bindings to your project. You can also use these bindings in development with [Wrangler](https://developers.cloudflare.com/pages/platform/functions#develop-and-preview-locally). 
```

### KV namespace

Workers KV is Cloudflare's globally replicated key-value storage solution. Within Pages, you can choose from the list of KV namespaces that you created from the dashboard by going to  **Account Home** > **Pages** > **your Pages project** > **Settings** > **Functions** > **KV namespace bindings**. Select **Add binding** and input a **Variable name** and select a *KV namespace* from the list of your existing Workers KV namespaces. You will need to repeat this for both the **Production** and **Preview** environments.

![KV-Binding](KV-functions.png)

> ## KV namespace Locally 

Reviewer: ```suggestion
## KV namespace locally
```

> While developing locally you can interact with your KV namespace by add `-k, --kv  [Namespace name]` to your run command. For example, if your namespace is called TodoList you can access the KV namespace in your local dev by running `npx wrangler pages dev dist  --kv TodoList`. The data from this namespace can be accessed using `context.env`. 

Reviewer: ```suggestion
While developing locally you can interact with your KV namespace by add `-k, --kv  [Namespace name]` to your run command. For example, if your namespace is called `TodoList`, you can access the KV namespace in your local dev by running `npx wrangler pages dev dist  --kv TodoList`. The data from this namespace can be accessed using `context.env`. 
```

```js
export async function onRequest({ env }) {
 await env.TodoList.put("Task", "Clean Kitchen")
  const out = await env.TodoList.get("Task");
  return new Response(out);
}
```

### Durable Object namespace

Durable Objects are Cloudflare's strongly consistent coordination primitive that power capabilities such as connecting WebSockets, handling state, and building applications. As with Workers KV, you first have to [create the Durable Object](https://developers.cloudflare.com/workers/learning/using-durable-objects#uploading-a-durable-object-worker). You can then configure it as a binding to your Pages project.

Go to **Account Home** > **Pages** > **your Pages project** > **Settings** > **Functions** > **Durable Object bindings**. Select **Add binding** and input a **Variable name** and select a *Durable Object namespace* from the list of your existing Durable Objects. You will need to repeat this for both the **Production** and **Preview** environments.

![DO-Binding](DO-functions.png)

## Durable Objects locally

> Just as you can access kv with `-k`or `-kv` you can access durable objects in your local builds with `-o`, `--do` followed by your Durable object name and class. 

Reviewer: ```suggestion
Access Durable Objects in your local builds with `-o`, `--do` followed by your Durable object name and class. Similarly, you can access KV with `-k`or `-kv`.
```

Author: @deadlypants1973 Since KV is mentioned before durable objects, I think it makes sense to say "Just as you can do ...... in kv, you can do ..... in DO". Reading by how they are introduced in the doc.

Reviewer: Rule of thumb is to put the instruction for what the header is introducing first. Is it necessary to have the bit about KV? I would recommend taking it out and just leaving the instructions for DOs here.

Author: Just trying to show similarity, since there isn’t an example to DO here. I think it'll be helpful when setting up DO with a Function example.

Author: Maybe I can reword it, what do you think of how it reads now?

![ENV-Binding](ENV-functions.png)

> ## Adding Environment variable locally 

Reviewer: ```suggestion
## Adding environment variables locally
```

> When developing in your local environment, functions offers you a way to use your enviroment varibles with `context.env` you can access environment varibles on build by adding a binding to your run command like `npx wrangler pages dev dist --binding ENV_NAME=\"ENV_VALUE"`. This allows you to then access the ENV_VALUE in your component by using `env.ENV_NAME`.

Reviewer: ```suggestion
When developing in your local environment, Functions offers you a way to use your environment variables with `context.env`. Access environment variables on your build by adding a binding to your run command like `npx wrangler pages dev dist --binding ENV_NAME=\"ENV_VALUE"`. This allows you to then access the ENV_VALUE in your component by using `env.ENV_NAME`.
```

> For example, you can connect [Sentry](https://www.sentry.io/) to your application middleware using [Toucan js](https://github.com/robertcepa/toucan-js) and access your dsn in your component. 

Reviewer: Should dsn be capitalized?

Author: Oh yes it should be. Sorry I wrote it how toucan accepts the options
